### PR TITLE
add clip_by_norm op

### LINKED ISSUE #4662: clip_by_norm operator.
- TF Doc: https://www.tensorflow.org/api_docs/python/tf/clip_by_norm
- PyTorch Doc: http://pytorch.org/docs/master/nn.html#torch.nn.utils.clip_grad_norm

### PR DESCRIPTION
**Pull request by wwhu**
fix #4662 
Clipping the gradient, only forward pass is needed.

### CODE REVIEW

File: paddle/operators/clip_by_norm_op.h [hunk 1]
```diff
@@ -0,0 +1,55 @@
...
+namespace paddle {
+namespace operators {
+
+using Tensor = framework::Tensor;
+template <typename T, int MajorType = Eigen::RowMajor,
+          typename IndexType = Eigen::DenseIndex>
+using EigenVector = framework::EigenVector<T, MajorType, IndexType>;
+template <typename T, int MajorType = Eigen::RowMajor,
+          typename IndexType = Eigen::DenseIndex>
+using EigenScalar = framework::EigenScalar<T, MajorType, IndexType>;
```
Comment (qingqing01) [inline paddle/operators/clip_by_norm_op.h]: If `EigenScalar ` is not used, it can be removed.

File: paddle/operators/clip_by_norm_op.cc [hunk 2]
```diff
@@ -0,0 +1,90 @@
...
+  void InferShape(framework::InferShapeContext* ctx) const override {
+    PADDLE_ENFORCE(ctx->HasInput("X"), "Input(X) should not be null");
+    PADDLE_ENFORCE(ctx->HasInput(framework::GradVarName("Out")),
+                   "Input(Out@GRAD) should not be null");
+    auto x_dims = ctx->GetInputDim("X");
+    if (ctx->HasOutput(framework::GradVarName("X"))) {
+      ctx->SetOutputDim(framework::GradVarName("X"), x_dims);
+    }
+  }
+};
```
Comment (qingqing01) [inline paddle/operators/clip_by_norm_op.cc]: If this Op has no gradient op, the `ClipByNormOpGrad` can be removed.

File: paddle/operators/clip_by_norm_op.cc [hunk 3]
```diff
@@ -0,0 +1,90 @@
...
+
+template <typename AttrType>
+class ClipByNormOpMaker : public framework::OpProtoAndCheckerMaker {
+ public:
+  ClipByNormOpMaker(framework::OpProto* proto,
+    framework::OpAttrChecker* op_checker)
+      : OpProtoAndCheckerMaker(proto, op_checker) {
+    AddInput("X",
+             "(Tensor)The input of clip_by_norm op."
```
[the remark this answers is not in the document]
Comment (wwhu) [reply paddle/operators/clip_by_norm_op.cc]: Space is added.

File: paddle/operators/clip_by_norm_op.cc (hunk 2 above)
Comment (wwhu) [reply paddle/operators/clip_by_norm_op.cc]: removed

File: paddle/operators/clip_by_norm_op.h (hunk 1 above)
Comment (wwhu) [reply paddle/operators/clip_by_norm_op.h]: removed
